File: src/main/java/com/example/librarymanagement/dto/mapper/LibraryMapper.java
package com.example.librarymanagement.dto.mapper;

import com.example.librarymanagement.dto.LibraryDto;
import com.example.librarymanagement.model.entity.Library;
import org.springframework.stereotype.Component;
import org.springframework.util.CollectionUtils;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@Component
public class LibraryMapper {
    public Library toLibrary(LibraryDto libraryDto) {
        Library library = new Library();
        library.setName(libraryDto.getName());
        library.setAddress(libraryDto.getAddress());
        return library;
    }

    public LibraryDto toLibraryDto(Library library) {
        LibraryDto libraryDto = new LibraryDto();
        libraryDto.setId(library.getId());
        libraryDto.setName(library.getName());
        libraryDto.setAddress(library.getAddress());
        return libraryDto;
    }

    public List<LibraryDto> toLibraryDto(List<Library> libraries) {
        if(CollectionUtils.isEmpty(libraries)) return Collections.emptyList();

        return libraries.stream()
                .map(this::toLibraryDto)
                .toList();
    }
}

File: src/main/java/com/example/librarymanagement/dto/mapper/LibraryMapper.kt
package com.example.librarymanagement.dto.mapper

import com.example.librarymanagement.dto.LibraryDto
import com.example.librarymanagement.model.domain.Library
import org.springframework.stereotype.Component

@Component
class LibraryMapper {
    fun toLibrary(libraryDto: LibraryDto, id: String? = null): Library {
        return Library(id = libraryDto.id ?: id, libraryDto.name, address = libraryDto.address)
    }

    fun toLibraryDto(library: Library): LibraryDto = LibraryDto(library.id!!, library.name, library.address)

    fun toLibraryDto(libraries: List<Library>): List<LibraryDto> = libraries.map { toLibraryDto(it) }
}
